compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -642,6 +643,11 @@ fn reg_to_llvm(reg: InlineAsmRegOrRegClass, layout: Option<&TyAndLayout<'tcx>>)
             InlineAsmRegClass::Wasm(WasmInlineAsmRegClass::local) => "r",
             InlineAsmRegClass::Bpf(BpfInlineAsmRegClass::reg) => "r",
             InlineAsmRegClass::Bpf(BpfInlineAsmRegClass::wreg) => "w",
+            InlineAsmRegClass::Avr(AvrInlineAsmRegClass::reg) => "r",
+            InlineAsmRegClass::Avr(AvrInlineAsmRegClass::reg_upper) => "d",
+            InlineAsmRegClass::Avr(AvrInlineAsmRegClass::reg_pair) => "r",
+            InlineAsmRegClass::Avr(AvrInlineAsmRegClass::wreg) => "w",
```

Comment:
The AVR LLVM backend knows 3 special pointer registers X, Y, Z (r25-r31 or so). This register class forms the "pointer register class"

I haven't seen the `W` register (I'm inferring it would be `r25r24`) mentioned in the instruction set docs anywhere. Obviously r25r24 still exists but I don't think it is conventional to give it the name `W` because the instruction set does not allow it to be used as an pointer so it's not really a special register deserving of an alphabetic alias.

Recommend removing `wreg` from AVR inline asm. Unless there's a reason/usecase I'm missing

Reply:
Ignore this comment, I just read the manual (properly this time) and wreg is an existing constraint and you have implemented the correct registers in it